code/game/objects/items/robot/robot_upgrades.dm

```diff
@@ -99,6 +99,12 @@
 	if (/obj/item/borg/upgrade/vtec in R.module.upgrades)
 		R.movement_speed_modifier -= vtec_bonus
 
+	for(var/obj/item/borg/upgrade/thing in R)
+		if(istype(thing, /obj/item/borg/upgrade/reset))
```

Comment:
would it make more sense to just have the reset upgrade never inserted in the first place? why track the object instead of deleting it when used?

Reply:
i'd have to snowflake upgrade code for that but you got a point.

Reply:
Wait no i can do that just fine